official/resnet/resnet_run_loop.py

```diff
@@ -86,21 +97,23 @@ def process_record_dataset(dataset, is_training, batch_size, shuffle_buffer,
   # last batch of an epoch-- do not raise an error when we try to split them
   # over the GPUs. This will likely be handled by Estimator during replication
   # in the future, but for now, we just drop the leftovers here.
-  if multi_gpu:
+  if use_distribution_strategy:
     total_examples = num_epochs * examples_per_epoch
-    dataset = dataset.take(batch_size * (total_examples // batch_size))
+    dataset = dataset.take(
+        per_device_batch_size * (total_examples // per_device_batch_size))
 
   # Parse the raw records into images and labels
-  dataset = dataset.map(lambda value: parse_record_fn(value, is_training),
-                        num_parallel_calls=num_parallel_calls)
-
-  dataset = dataset.batch(batch_size)
+  dataset = dataset.apply(
+      tf.contrib.data.map_and_batch(
+          lambda value: parse_record_fn(value, is_training),
+          batch_size=per_device_batch_size,
+          num_parallel_batches=1))
 
   # Operations between the final prefetch and the get_next call to the iterator
   # will happen synchronously during run time. We prefetch here again to
   # background all of the above processing work and keep it out of the
   # critical training path.
-  dataset = dataset.prefetch(1)
+  dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
```

Comment:
thanks!

Reply:
Please add instructional comments-- what is autotune? What does it do?